_posts/2019-03-12-so-you-want-to-be-an-engineer.md

```diff
@@ -0,0 +1,170 @@
+
+I wanted to work at Artsy, and I felt that I was most interested in front-end work. So I asked a few of our
+engineers to help me understand our stack and to recommend frameworks/languages I should learn.
+
+If you're interested in changing companies as well as careers, it's worth seeing if your target company or
+companies have open source code you can check out. See if you can find them on [GitHub](https://www.github.com) and
+look at some of their recently updated apps. If you already have some coding experience, see if you can contribute
+a little bit—even fixing small bugs or typos is a good place to start, since you're both contributing to their
+codebase (which will help you if you land an interview) and learning more about their stack. dB recommended that I
+do this with Artsy's code.
+
+As with most companies that have been around for more than a year or two, Artsy is home to projects with many
+different stacks. However, most of our newer front-end stuff—things that are recently updated or currently in
+development—is built on React using TypeScript. So my first question was "what do I need to know in order to write
+TypeScript code in a React framework?"
+
+Fortunately, a lot of other people have the same question.
+
+# Step Three: Decide How You Want to Learn It
+
+First: **there is no wrong way to go about learning to code**. Whether it takes you 6 months or 6 years, whether
+you learn one language or a dozen, whether you ask for a lot of help or do it all yourself, **you are learning and
+that is valuable**. Learning to code is not a magical skill. Like just about everything, it's a matter of putting
+in time. Just keep trying, even small things, and you _will_ make progress.
+
+Lots of people who want to become engineers go to coding bootcamps like [Flatiron](https://flatironschool.com/) or
+[General Assembly](https://generalassemb.ly/). That's a very reasonable thing to do! I had a decent grounding in CS
+fundamentals thanks to my experience in undergrad, and as a result, I initially decided I wouldn't do a coding
+bootcamp—I felt I had enough experience to benefit from the multitude of online courses and open source projects
+out there.
+
+However, there are definitely real and significant benefits do doing a bootcamp. Here are three that I can think
+of:
+
+1. Clear curriculum. At times, I was overwhelmed by the possible ways to proceed and stopped making progress as a
+   result. Bootcamps take the guesswork out.
+2. Timeline. Because I didn't have a hard date set for a transition or interview, it was up to me to determine how
+   fast/slow I worked on things, and I stagnated at times as a result.
+3. Community. While the open source community is very much a real thing, and you can find lovely people on the
+   internet who want to help learners like yourself, it's not the same as having multiple people learning the same
+   thing at the same time in the same physical space.
```

Comment:
These three points are very well-constructed – you point out the benefits and how it applied to you, which is the perfect tone for this post 👍

Reply:
thank you!